hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -784,29 +788,103 @@ private AbfsOutputStreamContext populateAbfsOutputStreamContext(
             .build();
   }
 
-  public void createDirectory(final Path path, final FsPermission permission,
+  public void createDirectory(final Path path, final FileSystem.Statistics statistics, final FsPermission permission,
```

Comment:
no need to have statistics here. Reason being, it just a createPath we want to have. We do not want to write anything. Even if the AbfsOutputStream generated would not have `statistics`, it doesn't matter, because when using it, there is null check: https://github.com/ABFSDriver/AbfsHadoop/blob/8e5620cd9e58f39dafc3c16edaf7f6d6f9d4af79/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java#L412-L414. This will come only if there outputStream.write() happen, which will not happen here. Because, as part of createDirectory, its just putBlob API which will be called.

Reply:
Don't want to pass down null parameter for statistics, want to be inline with what code we would have called if the check was in filesystem class. Hence adding the parameter. The createDirectory function has only one usage across all classes so should not be an issue passing this down.